lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -170,19 +175,29 @@ def initialize(options = {})
         super
       end
 
-      def authorize(money, creditcard_or_reference, options = {})
-        setup_address_hash(options)
-        commit(build_auth_request(money, creditcard_or_reference, options), :authorize, money, options)
+      def authorize(money, payment_method, options = {})
+        if valid_payment_method?(payment_method)
+          setup_address_hash(options)
+          commit(build_auth_request(money, payment_method, options), :authorize, money, options)
+        else
+          # this is for NetworkToken brands that aren't supported at CyberSource
+          Response.new(false, "Payment method #{card_brand(payment_method)} is not supported, check https://developer.cybersource.com/docs/cybs/en-us/payments/developer/all/rest/payments/CreatingOnlineAuth/CreatingAuthReqPNT.html")
```

Comment:
Should we clarify in this response message that it is for network tokens? Or will the return value from `card_brand(payment_method)` make that clear?

Reply:
That's a good idea, I'll update the error message since I've added this to include ApplePay and GooglePay. I decided to fix this since currently we're getting `The payment gateway settings are most likely invalid.` which is very misleading.